crates/assistant/src/assistant_panel.rs

```diff
@@ -1507,7 +1508,9 @@ impl Conversation {
             )
             .await,
         );
-        cx.update(|cx| completion_provider.retrieve_credentials(cx))?;
+        cx.update(|cx| completion_provider.retrieve_credentials(cx))?
+            .await;
```

Comment:
We weren't `await`ing the call to retrieve the credentials 🤦🏻‍♂️

Reply:
Would `#[must_use]` help if placed on `BoxFuture` & friends?

Reply:
I was wondering that.

It looks like `BoxFuture` is coming from the `futures` crate, so we'd either have to wrap it or try to upstream.

Reply:
I've done a blasphemous direct update of `.cargo/registry/src` file with the `BoxFuture` and it does not seem to help, alas: with or without this `.await`, the `cargo clean & cargo check --all --all-targets` output did not change.

Reply:
And as @\osiewicz noted, `cargo +nightly check` does detect such issues: it seems there was exactly one place without `.await` in the project currently.